### PR TITLE
generate-bootstraps.sh: adding a way to generate bootstrap based on pacman

### PR DESCRIPTION
**Pull request by Maxython**
#10256

So far only done for `generate-bootstraps.sh`.  Already tested everything, it works fine.  I'll work on improving `build-bootstraps.sh` later.

### CODE REVIEW

**Comment by Maxython** [comment]
Unfortunately I can't do development on `build-bootstraps.sh`.  I think it would be right to make these requests for consideration, and create them separately for `build-bootstraps.sh`.

**Comment by Maxython** [comment]
@agnostic-apollo

File: scripts/generate-bootstraps.sh [hunk 1]
```diff
@@ -206,6 +285,9 @@ show_usage() {
 	echo "                             Multiple packages should be passed as"
 	echo "                             comma-separated list."
 	echo
+	echo " --pm                        Set up a package manager in bootstrap."
```
Comment (agnostic-apollo) [inline scripts/generate-bootstraps.sh:288]: Change to `--pm MANAGER`

File: scripts/generate-bootstraps.sh [hunk 2]
```diff
@@ -243,6 +326,24 @@ while (($# > 0)); do
 				exit 1
 			fi
 			;;
+		--pm)
+			case "$2" in
+				"apt"|"pacman")
+					TERMUX_PACKAGE_MANAGER="$2"
+					if [ ${TERMUX_PACKAGE_MANAGER} = "apt" ]; then
+						REPO_BASE_URL="https://packages.termux.org/apt/termux-main"
```
Comment (agnostic-apollo) [inline scripts/generate-bootstraps.sh:334]: Define all urls at start of file instead of duplicates.

**Comment by agnostic-apollo** [comment]
Otherwise looks fine from a quick overview. For pacman issues, you would need to solve yourself since I am not familiar with it or its build system.

>Unfortunately I can't do development to build-bootstraps.sh.

Yeah, not a necessity. Up to you and other pacman users.

**Comment by agnostic-apollo** [comment]
Can you allow edits for maintainers?

**Comment by Maxython** [comment]
> Can you allow edits from maintainers?

That is, to allow you to edit the file?
A little, I did not understand your question.

**Comment by agnostic-apollo** [comment]
Yeah, there should be an option in the side bar of the pull request for you.

**Comment by Maxython** [comment]
> Yeah, there should be an option in the side bar of the pull request for you.

I don't have that option.

**Comment by Grimler91** [comment]
@agnostic-apollo I think you should have access, normally we are able to force push to the remote branch ([termux-pacman:bootstraps](https://github.com/termux-pacman/termux-packages/tree/bootstraps) in this case) in the PR, unless the branch is protected or something in the other repo

**Comment by agnostic-apollo** [comment]
I am getting

```
ERROR: Permission to termux-pacman/termux-packages.git denied to agnostic-apollo.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
```

Access is granted with `Allow edits from maintainers` toggle when you a create pull request or after that too. But since `termux-pacman` is an organization and not a user account, that toggle won't exist.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Anyways, undo your last commit with `git reset --hard HEAD^` and apply patch with `git am < git.txt`

[git.txt](https://github.com/termux/termux-packages/files/8571694/git.txt)

**Comment by Grimler91** [comment]
Yeah, you are right, same issue in https://github.com/termux/termux-packages/pull/10359

**Comment by Maxython** [comment]
@agnostic-apollo thank you. 

I will probably give you access to the termux-packages repo so that this situation does not happen again.

**Comment by agnostic-apollo** [comment]
You are welcome and thanks to you too.

That can be done, although not currently necessary. :)

**Pull request merged**